integration-tests/paho/src/main/java/org/apache/camel/quarkus/component/paho/PahoResource.java

```diff
@@ -87,4 +88,11 @@ public String mqttExceptionDuringReconnectShouldSucceed() {
         return mqex.getMessage();
     }
 
+    @Path("/readThenWriteWithFilePersistenceShouldSucceed")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String readThenWriteWithFilePersistenceShouldSucceed(@QueryParam("message") String message) throws Exception {
+        producerTemplate.requestBody("paho:withFilePersistence?retained=true&cleanSession=false&persistence=FILE", message);
```

Comment:
if I remove ```cleanSession=false``` , the native test starts to fail but the jvm test is OK. I just wonder if it is a workaround for the native test or it is a limitination?

Reply:
Indeed, I have the same with mqtt3. In native with cleanSession = true, there is a NullPointerException. So far I've been able to investigate at this stage, it appears that the clientDir is deleted.

The deletion show up only in native mode, but I suspect it is a more general race condition. Perharps even in camel-paho itself. At this stage, I was not able to fully qualify this bug. cleanSession = false allow to test with file persistence without this bug.

Reply:
yeah, the similar situation with mqtt5 but I have to add ```cleanStart=false```. I took some debug works by adding some ```system.out.println``` and [clientDir.delete()](https://github.com/eclipse/paho.mqtt.java/blob/master/org.eclipse.paho.mqttv5.client/src/main/java/org/eclipse/paho/mqttv5/client/persist/MqttDefaultFilePersistence.java#L298) return ```false``` in jvm mode but ```true``` in native mode.

It might be worth to raise a upstream CAMEL issue track this problem.

Reply:
Hey, great catch. Would you by chance be able to check whether the `clientDir` is empty before deletion in native and JVM ? It might be a reason to return false.

Tracking this issue is definitely a good idea. Should the issue be opened in camel-quarkus or camel, I don't know. What do you think ?

Reply:
Ok, that would be a threading issue that we just see more often in native mode then. Really nice @zhfeng :+1:

Reply:
Ha, intersting ! ```java.nio.file.Files.delete(Path)``` throws ```java.nio.file.DirectoryNotEmptyException```. There is ```.lck``` in  the directory. It looks like the client still holds a ```FileLock``` when clear() method is invoked.

Reply:
A theroy could be that the lock is not being released (in native mode only then):
https://github.com/eclipse/paho.mqtt.java/blob/master/org.eclipse.paho.mqttv5.client/src/main/java/org/eclipse/paho/mqttv5/client/internal/FileLock.java#L76

Maybe, I messed up the reflection config and `fileLock.getClass().getMethod("release",new Class[]{});` is throwing ?

Reply:
@aldettinger The root cause is we are missing ```RandomAccessFile``` reflection config which fails to get the file lock.
So adding
```
p.produce(new ReflectiveClassBuildItem(true, false, RandomAccessFile.class));
```
resloves the problem.

Reply:
Oh, great catch. I take a note to test this with mqtt3 too :)